src/watertap_contrib/reflo/analysis/case_studies/KBHDP/components/FPC.py

```diff
@@ -0,0 +1,231 @@
+from pyomo.environ import (
+    ConcreteModel,
+    value,
+    assert_optimal_termination,
+    units as pyunits,
+    Block,
+    Constraint,
+    SolverFactory,
+)
+import os
+
+from idaes.core import FlowsheetBlock, UnitModelCostingBlock
+from idaes.core.solvers import get_solver
+
+from watertap.core.util.model_diagnostics.infeasible import *
+from idaes.core.util.scaling import *
+
+from watertap_contrib.reflo.solar_models.surrogate.flat_plate.flat_plate_surrogate import (
+    FlatPlateSurrogate,
+)
+
+from idaes.core.util.model_statistics import (
+    degrees_of_freedom,
+    number_variables,
+    number_total_constraints,
+    number_unused_variables,
+)
+
+from watertap_contrib.reflo.costing import (
+    EnergyCosting,
+)
+
+__all__ = [
+    "build_fpc",
+    "init_fpc",
+    "set_fpc_op_conditions",
+    "add_fpc_costing",
+    "report_fpc",
+    "report_fpc_costing",
+]
+
+
+def build_system():
+    m = ConcreteModel()
+    m.fs = FlowsheetBlock(dynamic=False)
+    m.fs.costing = EnergyCosting()
+
+    m.fs.system_capacity = Var(initialize=6000, units=pyunits.m**3 / pyunits.day)
+
+    m.fs.fpc = FlowsheetBlock(dynamic=False)
+
+    return m
+
+
+def build_fpc(blk, __file__=None):
```

Comment:
@MuktaHardikar curious why you use a dunder here?

Reply:
I am also curious. Also nothing wrong with just putting the flat plate surrogate file right in the KBHDP case study directory since (presumably) it was developed for the case study.